File: src/server/agents/CodexCliAgent.ts
export class CodexCliAgent {
    public sandboxMode: string = "workspace-write";
    public reasoningMode: boolean = false;

    constructor() {}

    public enableO1Reasoning(enabled: boolean): void {
        this.reasoningMode = enabled;
        console.log(`O1 Reasoning mode set to: ${enabled}`);
    }

    public setSandboxMode(mode: string): void {
        this.sandboxMode = mode;
        console.log(`Sandbox mode set to: ${mode}`);
    }

    public async requestUserApproval(action: string): Promise<boolean> {
        console.log(`[TUI Prompt] User approval required for: ${action}`);

        return new Promise((resolve) => {
            setTimeout(() => {
                if (this.sandboxMode === "read-only") {
                    console.log("[TUI] Action denied by read-only sandbox");
                    resolve(false);
                } else {
                    console.log("[TUI] Action approved");
                    resolve(true);
                }
            }, 300);
        });
    }
}

File: java/src/main/java/com/maestro/agent/CodexCliAgent.java
package com.maestro.agent;

import java.util.concurrent.CompletableFuture;

public class CodexCliAgent {
    private String sandboxMode = "workspace-write";
    private boolean reasoningMode = false;

    public void enableO1Reasoning(boolean enabled) {
        this.reasoningMode = enabled;
        System.out.println("O1 Reasoning mode set to: " + enabled);
    }

    public void setSandboxMode(String mode) {
        this.sandboxMode = mode;
        System.out.println("Sandbox mode set to: " + mode);
    }

    public CompletableFuture<Boolean> requestUserApproval(String action) {
        return CompletableFuture.supplyAsync(() -> {
            System.out.println("[TUI Prompt] User approval required for: " + action);
            try { Thread.sleep(300); } catch (InterruptedException e) {}

            if ("read-only".equals(this.sandboxMode)) {
                System.out.println("[TUI] Action denied by read-only sandbox");
                return false;
            }

            System.out.println("[TUI] Action approved");
            return true;
        });
    }
}
